sdk/openai/openai-assistants/src/models/helpers.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * THIS IS AN AUTO-GENERATED FILE - DO NOT EDIT!
+ *
+ * Any changes you make here may be lost.
+ *
+ * If you need to make changes, please do so in the original source file, \{project-root\}/sources/custom
+ */
+import { RequiredToolCallOutput, ToolCallOutput } from "../rest/outputModels.js";
+import { RequiredToolCall, ToolCall } from "./models.js";
+
+export function parseRequiredToolCallOutput(requiredToolCallOutput: RequiredToolCallOutput): RequiredToolCall {
+  return { type: "function", id: requiredToolCallOutput.id, function: requiredToolCallOutput.function } as RequiredToolCall;
+}
+
+export function parseToolCallOutput(toolCallOutput: ToolCallOutput): ToolCall {
+  const { id } = toolCallOutput;
+  const toolCall: {
+    id: string;
+    type: string;
+    function: any;
+    retrieval: any;
+    codeInterpreter: any;
+  } = { id, type: "", function: {}, retrieval: {}, codeInterpreter: {} };
+  switch (toolCallOutput.type) {
+    case "function":
+      toolCall.type = toolCallOutput.type as "function";
+      toolCall.function = toolCallOutput.function;
+      break;
+    case "retrieval":
+      toolCall.type = toolCallOutput.type as "retrieval";
+      toolCall.retrieval = toolCallOutput.retrieval;
+      break;
+    case "code_interpreter":
+      toolCall.type = toolCallOutput.type as "code_interpreter";
+      toolCall.codeInterpreter = toolCallOutput.code_interpreter;
+      break;
+    default:
+      throw new Error(`Unknown tool call type: ${toolCall.type}`);
+  }
+
+  return toolCall as ToolCall;
```

Comment:
nitpick: I think this can be simplified as follows:
```suggestion
  const { id, type } = toolCallOutput;
  switch (type) {
    case "function":
      return { type, function: toolCallOutput.function };
    case "retrieval":
      return { type, retrieval: toolCallOutput.retrieval };
    case "code_interpreter":
      return { type, codeInterpreter: toolCallOutput.code_interpreter };
    default:
      throw new Error(`Unknown tool call type: ${type}`);
  }
```

Reply:
Or by just using the camel case renaming helper.

Reply:
[Updated](https://github.com/Azure/azure-sdk-for-js/pull/27807/commits/0c67835b91312c9b47bfbb2fdd83fa7c49fb66d2)